### PR TITLE
Pr vaapi vp9 drc

### PR DESCRIPTION
**Pull request by fulinjie**
Fix vp9 dynamic resolution change clips with an in-place re-init function without destroy anything.

### CODE REVIEW

File: libavcodec/hwaccel.h [hunk 1]
```diff
@@ -24,6 +24,7 @@
 
 
 #define HWACCEL_CAP_ASYNC_SAFE      (1 << 0)
+#define HWACCEL_CAP_INTERNAL_ALLOC  (1 << 1)
```
Comment (xuguangxin) [inline libavcodec/hwaccel.h:27]: better change to a better name like HWACCEL_CAP_ FORMAT_CHANGE_ON_THE_FLY and add some comment.

File: libavcodec/decode.c [hunk 2]
```diff
@@ -1395,6 +1442,15 @@ int ff_get_format(AVCodecContext *avctx, const enum AVPixelFormat *fmt)
     memcpy(choices, fmt, (n + 1) * sizeof(*choices));
 
     for (;;) {
+        if (avctx->internal->hwaccel_priv_data &&
```
Comment (xuguangxin) [inline libavcodec/decode.c:1458]: better change this to a function named support_on_fly_formant_change()

File: libavcodec/decode.c [hunk 3]
```diff
@@ -1395,6 +1442,15 @@ int ff_get_format(AVCodecContext *avctx, const enum AVPixelFormat *fmt)
...
+            avctx->hwaccel->caps_internal & HWACCEL_CAP_INTERNAL_ALLOC) {
+            err = hwaccel_realloc_surface(avctx);
+            if (err < 0)
+                av_log(avctx, AV_LOG_WARNING, "Try to re-initialize all.\n");
```
Comment (xuguangxin) [inline libavcodec/decode.c:1462]: change to a better messeage. "on fly format change failed, Try to re-intit...."
But is this really usefful? If we reach here we do not have hw_frames_ctx, who will allocate new one for us?

File: libavcodec/decode.c [hunk 4]
```diff
@@ -1367,6 +1367,53 @@ static void hwaccel_uninit(AVCodecContext *avctx)
...
+    int ret;
+
+    if (avctx->hw_frames_ctx)
+        av_buffer_unref(&avctx->hw_frames_ctx);
+
+    hw_config = get_hw_config(avctx, avctx->pix_fmt);
+    if (!hw_config)
+        return AV_PIX_FMT_NONE;
+
+    if (avctx->hw_device_ctx &&
```
Comment (xuguangxin) [inline libavcodec/decode.c]: there are many if at https://github.com/intel-media-ci/ffmpeg/pull/153/files#diff-ea4f4194be6309231a5b01e350c272cdL1448, why we only handle this?

**Comment by fulinjie** [comment]
closed,
some details in:
https://docs.google.com/document/d/1_fbhj0ik8Ir0S4eH-b8WI_LrLyCdSyauUydYQe_1PxE/edit

**Pull request closed without merging**